src/libsyntax/codemap.rs

```diff
@@ -364,6 +364,46 @@ impl CodeMap {
         }
     }
 
+    /// Returns `Some(span)`, a union of the lhs and rhs span.  The lhs must precede the rhs. If
+    /// there are gaps between lhs and rhs, the resulting union will cross these gaps.
+    /// For this to work, the spans have to be:
+    ///    * the expn_id of both spans much match
+    ///    * the lhs span needs to end on the same line the rhs span begins
+    ///    * the lhs span must start at or before the rhs span
```

Comment:
Do you expect spans to overlap sometimes? The only time I would expect that is if one is strictly larger than the other, in which case maybe this method shouldn't be used. If you agree, you could be stricter and require the `sp.lhs.hi < sp.rhs.lo`, rather than comparing both `lo`s